oauth2/src/main/java/com/okta/spring/boot/oauth/OktaOAuth2Configurer.java

```diff
@@ -110,4 +128,22 @@ private void configureResourceServer(HttpSecurity http, OktaOAuth2Properties okt
 
         return accessTokenResponseClient;
     }
+
+    private boolean isOpaqueTokenValidationRequired(OktaOAuth2Properties oktaOAuth2Properties,
+                                                    OAuth2ResourceServerProperties resourceServerProperties) {
+        return oktaOAuth2Properties.isOpaque()
+            && !isEmpty(resourceServerProperties.getOpaquetoken().getClientId())
+            && !isEmpty(resourceServerProperties.getOpaquetoken().getClientSecret())
+            && !isEmpty(resourceServerProperties.getOpaquetoken().getIntrospectionUri());
+    }
+
+    private boolean isRootOrgIssuer(String issuerUri) {
+        try {
+            ConfigurationValidator.assertOrgUrl(issuerUri);
```

Comment:
Fine for debugging, but this has the potential to fail in other was, for example, if the issuer URL was non-https.

Reply:
a non-https issuer URL will yield a `false` return value from this method (https check is handled [here](https://github.com/okta/okta-commons-java/blob/master/config-check/src/main/java/com/okta/commons/configcheck/ConfigurationValidator.java#L251) by the library).

Reply:
Sorry, right bad example. but detecting if a URL is a root issuer or not shouldn't be done this way as that isn't the intent of this method.  My suggestion is to check if the URL contains something like `/oauth/v1` (or something), if not, it's a root org issuer.  This also prevents using the exception handling for the normal expected case.

Reply:
So if we detect root org issuer, we should enforce Opaque Token introspection flow, right? If so,  how should we setup the introspection URI? Presently it is set to a value relative to the issuer URL (`${okta.oauth2.issuer}/v1/introspect`) - is that right? or should we let the user supply the introspection uri?.

Reply:
Yes, making it relative to the issuer URL.  The user should NOT need to supply the introspection URL manually.  In this case, we can hard code it because it's not going to change.

Reply:
addressed